pandas/core/indexes/period.py

```diff
@@ -320,7 +320,9 @@ def _is_comparable_dtype(self, dtype: DtypeObj) -> bool:
         freq = dtype.freq
         own_freq = self.freq
         return (
-            freq._period_dtype_code == own_freq._period_dtype_code
+            freq._period_dtype_code
+            # error: "BaseOffset" has no attribute "_period_dtype_code"
+            == own_freq._period_dtype_code  # type: ignore[attr-defined]
```

Comment:
i think this is a perf optimization for `dtype == self.dtype` that might no longer be necessary; i think we do this in PeriodDtype.__eq__ now.

Reply:
Should probably be done in a separate PR.